### ISSUE TITLE
MediaCodecVideoRenderer when reuse player decoder

### ISSUE THREAD

**Issue by Hiwensen** [google/ExoPlayer#10572]
Hi, 
I was following this [blog](https://medium.com/google-exoplayer/improved-decoder-reuse-in-exoplayer-ef4c6d99591d), trying to reuse the decoder for the playback. Skip call ExoPlayer.release() after each playback, just call ExoPlayer.prepare() when playing a new title.
But I met an error when starting to play a new video(set a new video URL), didn't repro the error when playing the same video,
and didn't repro the error when calling ExoPlayer.release() every time exit the playback.

* I didn't use PlayerView provided by ExoPlayer. I added a SurfaceView to the `AspectRatioFrameLayout`, just like `PlayerView` does, but I do reuse the `SurfaceView` for all the playback.
* The error happened when playing DRM(widevine) resources, didn't happen when playing non-DRM contents
* The error is not related to a piece of media.
* It can be reproduced on a Pixel 6 device running android 12
* ExoPlayer version 2.16.1

I've been debugging for a few days trying to figure out where the problem is but with no progress. 
Can you provide some information that might be causing this error？
Thanks in advance.

Below is the error detail info and the bug report.
```

2022-08-31 11:21:16.294 24285-24753/com.t.dev E/BufferQueueProducer: [SurfaceView[com.t.dev/com.t.activities.MainActivity]#6(BLAST Consumer)6](id:5edd0000000a,api:3,p:24285,c:24285) detachBuffer: slot 62 is not owned by the producer (state = FREE)
2022-08-31 11:21:16.295 24285-24753/com.t.dev E/BufferQueueProducer: [SurfaceView[com.t.dev/com.t.activities.MainActivity]#6(BLAST Consumer)6](id:5edd0000000a,api:3,p:24285,c:24285) detachBuffer: slot 63 is not owned by the producer (state = FREE)
2022-08-31 11:21:16.305 24285-24516/com.t.dev E/BufferQueueProducer: [MediaCodec.release](id:5edd0000000b,api:0,p:-1,c:24285) detachBuffer: BufferQueue has no connected producer
2022-08-31 11:21:26.502 24285-24522/com.t.dev E/ExoPlayerImplInternal: Playback error
      com.google.android.exoplayer2.ExoPlaybackException: MediaCodecVideoRenderer error, index=0, format=Format(2, null, null, video/avc, avc1.640028, 3175000, null, [1280, 720, -1.0], [-1, -1]), format_supported=YES
        at com.google.android.exoplayer2.ExoPlayerImplInternal.handleMessage(ExoPlayerImplInternal.java:562)
        at android.os.Handler.dispatchMessage(Handler.java:102)
        at android.os.Looper.loopOnce(Looper.java:201)
        at android.os.Looper.loop(Looper.java:288)
        at android.os.HandlerThread.run(HandlerThread.java:67)
     Caused by: android.media.MediaCodec$CryptoException: Attempted to use a closed session
        at android.media.MediaCodec.native_queueInputBuffer(Native Method)
        at android.media.MediaCodec.queueInputBuffer(MediaCodec.java:2634)
        at com.google.android.exoplayer2.mediacodec.AsynchronousMediaCodecBufferEnqueuer.doQueueInputBuffer(AsynchronousMediaCodecBufferEnqueuer.java:226)
        at com.google.android.exoplayer2.mediacodec.AsynchronousMediaCodecBufferEnqueuer.doHandleMessage(AsynchronousMediaCodecBufferEnqueuer.java:199)
        at com.google.android.exoplayer2.mediacodec.AsynchronousMediaCodecBufferEnqueuer.access$000(AsynchronousMediaCodecBufferEnqueuer.java:47)
        at com.google.android.exoplayer2.mediacodec.AsynchronousMediaCodecBufferEnqueuer$1.handleMessage(AsynchronousMediaCodecBufferEnqueuer.java:96)
        at android.os.Handler.dispatchMessage(Handler.java:106)
        at android.os.Looper.loopOnce(Looper.java:201) 
        at android.os.Looper.loop(Looper.java:288) 
        at android.os.HandlerThread.run(HandlerThread.java:67) 
2022-08-31 11:21:26.515 24285-24807/com.t.dev E/BufferQueueProducer: [SurfaceView[com.t.dev/com.t.activities.MainActivity]#8(BLAST Consumer)8](id:5edd0000000d,api:3,p:24285,c:24285) detachBuffer: slot 0 is not owned by the producer (state = FREE)
2022-08-31 11:21:26.516 24285-24807/com.t.dev E/BufferQueueProducer: [SurfaceView[com.t.dev/com.t.activities.MainActivity]#8(BLAST Consumer)8](id:5edd0000000d,api:3,p:24285,c:24285) detachBuffer: slot 1 is not owned by the producer (state = FREE)

```

[bugreport-oriole-SP2A.220305.013.A3-2022-08-31-11-29-55.zip](https://github.com/google/ExoPlayer/files/9458205/bugreport-oriole-SP2A.220305.013.A3-2022-08-31-11-29-55.zip)

**Comment by tonihei** [comment]
I just did a very simple test to call `player.setMediaItem(mediaItemWithDRM)` after some delay during playback of DRM-protected content and didn't see a problem:
```
new Handler(Looper.myLooper()).postDelayed(() -> {
      player.setMediaItems(mediaItems, /* resetPosition= */ true);
    },20000);
```

Does your issue reproduce in the demo app? Based on your description it should be easy to make a small modification similar to the one above to reproduce it. Could you try this and describe the changes needed to trigger the issue?

**Comment by Hiwensen** [comment]
> 

Thanks for your response!
Will try and get back to you later.

**Comment by Hiwensen** [comment]
@tonihei This issue is not reproduced in the demo app, I added the code you posted, and neither repro it.
I was trying to reuse the SurfaceView when met the error, which is to use the same SurfaceView for multiple playback sessions.
Do you know under what circumstances does the error I posted happen?

**Comment by Hiwensen** [comment]
@tonihei 
I reproduced the error on ExoPlayer demo app with tag `r2.16.1` after adding the code below to `PlayerActivity.initializePlayer()`, can't repro the error on version `2.17.0` and later.
Could you help confirm whether the error is fixed at https://github.com/google/ExoPlayer/issues/8696?

```
new Handler(Looper.getMainLooper()).postDelayed(() -> {
      final MediaItem mediaItem = new MediaItem.Builder()
          .setUri("https://test-videos.co.uk/vids/bigbuckbunny/mp4/h264/720/Big_Buck_Bunny_720_10s_2MB.mp4")
          .build();
      player.setMediaItem(mediaItem);
      player.prepare();
    }, 20000);
```

**Comment by tonihei** [comment]
Yes, #8696 could be the same issue or the fix of #8696 could just address this problem as well. 

@icbaker Assigning to you in case you want to follow/up or investigate further. But I assume we don't need to do anything if the issue is already fixed in the latest release.

**Comment by Hiwensen** [comment]
@tonihei @icbaker Thanks! 
Close this issue for now since the error after switching from DRM content to clear content is fixed and I can't repro the error when playing DRM content with reusing decoder.
